src/inference/src/dev/make_tensor.cpp

```diff
@@ -269,7 +269,10 @@ class AllocatedTensor : public ViewTensor {
           m_allocator{allocator} {}
 
     ~AllocatedTensor() {
-        destroy_memory();
+        try {
```

Comment:
Is just more like work around to silent Coverity the better option is to have version `destroy_memory` which throw or not throw at all if exception is not possible

Reply:
If we get an exception from an allocator in dtor, we have 2 options: ignore it and try to continue to work or do nothing and let app to crash :)

Reply:
Our DefaultAllocator is potentially exception safe but user can specify any allocator on their own.
So I changed requirements to allocators. Now it has to have noexcept deallocate method